core/src/main/java/org/infinispan/transaction/impl/AbstractCacheTransaction.java

```diff
@@ -183,15 +184,9 @@ public void notifyOnTransactionFinished() {
    }
 
    @Override
-   public boolean waitForLockRelease(Object key, long lockAcquisitionTimeout) throws InterruptedException {
-      if (txComplete) return true; //using an unsafe optimisation: if it's true, we for sure have the latest read of the value without needing memory barriers
-      final boolean potentiallyLocked = hasLockOrIsLockBackup(key);
-      if (trace) log.tracef("Transaction gtx=%s potentially locks key %s? %s", tx, key, potentiallyLocked);
-      if (potentiallyLocked) {
-         notifier.await(lockAcquisitionTimeout, TimeUnit.MILLISECONDS);
-         return txComplete;
-      }
-      return true;
+   public final boolean waitForLockRelease(long lockAcquisitionTimeout) throws InterruptedException {
+      notifier.await(lockAcquisitionTimeout, TimeUnit.MILLISECONDS);
```

Comment:
Does the exception/return value support add a lot of overhead? Otherwise I'd rather standardize on CF everywhere instead of a custom synchronization primitive.